back/app/services/participation_permissions_service.rb

```diff
@@ -90,8 +98,13 @@ def commenting_disabled_reason_for_idea(idea, user)
   end
 
   def commenting_idea_disabled_reason_for_project(project, user)
-    phase = get_current_phase project
-    commenting_idea_disabled_reason_for_phase phase, user
+    project_visible_reason = project_visible_disabled_reason(project, user)
+    if project_visible_reason
+      project_visible_reason
+    else
+      phase = get_current_phase project
+      commenting_idea_disabled_reason_for_phase phase, user
```

Comment:
I wonder if at some point we wouldn't want to have action descriptors in the phase serializer. In that case, we might want to check for project visibility in `commenting_idea_disabled_reason_for_phase`. But not necessary at the moment.